File: backend/app/routers/audit_log.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.database import get_db
from app.schemas.audit_log import AuditLogResponse
from app.services.audit_log import get_all_audit_logs, get_audit_log_by_id
from app.core.deps import get_current_user
from app.models import User
from datetime import datetime
from app.schemas.pagination import PaginateResponse

router = APIRouter(prefix="/audit-logs", tags=["audit-logs"])

@router.get(
    "", 
    response_model=PaginateResponse[AuditLogResponse]
)
def get_audit_logs(
    user: User = Depends(get_current_user), 
    db: Session = Depends(get_db), 
    action: str | None = None, 
    resource: str | None = None, 
    resource_id: int | None = None, 
    status_code: int | None = None, 
    performed_by: int | None = None, 
    start: datetime | None = None, 
    end: datetime | None = None, 
    page: int = 1, 
    size: int = 20, 
):
    return get_all_audit_logs(user, db, action, resource, resource_id, status_code, performed_by, start, end, page, size)

@router.get(
    "/{id}", 
    response_model=AuditLogResponse
)
def get_audit_log(
    id: int,
    user: User = Depends(get_current_user), 
    db: Session = Depends(get_db), 
):
    return get_audit_log_by_id(id, user, db)

File: frontend/app/stores/audit-log.ts
import { type AuditLogResponse } from '~/types/audit-log'
import { type PaginateResponse } from '~/types/pagination'

export const useAuditLogStore = defineStore('audit-log', () => {
    const auditLogs = ref<AuditLogResponse[] | null>(null)    
    const total = ref(0)
    const currentAuditLog = ref<AuditLogResponse | null>(null)    
    const api = useApi()
    const fetchAuditLogs = async ({
        page = 1,
        size = 20,
        action, 
        resource, 
        resource_id, 
        status_code, 
        performed_by, 
        start, 
        end,
    }: {
        page?: number,
        size?: number,
        action?: string, 
        resource?: string, 
        resource_id?: number, 
        status_code?: number, 
        performed_by?: number, 
        start?: string, 
        end?: string,
    } = {}) => {
        const response: PaginateResponse<AuditLogResponse> = await api('/audit-logs', { params: { page, size, action, resource, resource_id, status_code, performed_by, start, end }})
        auditLogs.value = response.data
        total.value = response.total
    }
    const fetchAuditLog = async (id: number) => {        
        const response: AuditLogResponse = await api(`/audit-logs/${id}`)
        currentAuditLog.value = response
    }

    return {
        auditLogs, currentAuditLog, total, 
        fetchAuditLogs, fetchAuditLog
    }
})